sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -62,14 +64,13 @@ _nib_onl_entry_t *_nib_onl_alloc(const ipv6_addr_t *addr, unsigned iface)
 {
     _nib_onl_entry_t *node = NULL;
 
-    assert(addr != NULL);
     DEBUG("nib: Allocating on-link node entry (addr = %s, iface = %u)\n",
           ipv6_addr_to_str(addr_str, addr, sizeof(addr_str)), iface);
     for (unsigned i = 0; i < GNRC_IPV6_NIB_NUMOF; i++) {
         _nib_onl_entry_t *tmp = &_nodes[i];
 
         if ((_nib_onl_get_if(tmp) == iface) &&
-            (ipv6_addr_equal(addr, &tmp->ipv6))) {
+            ((addr == NULL) || (ipv6_addr_equal(addr, &tmp->ipv6)))) {
```

Comment:
The `@pre` is wrong. For prefix list entries the next-hop address may be `NULL`.